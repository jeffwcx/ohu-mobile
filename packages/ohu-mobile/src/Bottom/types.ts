export interface BottomProps {
  tag?: string;
  visible?: boolean;
  animation?: string;
  divider?: boolean;
}

export interface BottomEvents {
  onVisibleChange: boolean;
}
