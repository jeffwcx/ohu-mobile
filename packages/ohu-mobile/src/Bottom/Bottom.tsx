import { defineComponent, props } from '../_utils/defineComponent';
import { BottomProps } from './types';
import { $prefix } from '../_config/variables';
import Divider from '../Divider';

const Bottom = defineComponent<BottomProps>('bottom').create({
  props: {
    visible: props(Boolean).default(true),
    tag: props(String).default('div'),
    divider: props(Boolean).default(false),
    animation: props(String).default('slide-down'),
  },
  render(h) {
    const { $slots, tag, visible, animation, divider } = this;
    return (
      <transition name={`${$prefix}${animation}`}>
        {h(
          tag,
          {
            class: this.$rootCls(),
            directives: [{ name: 'show', value: visible }],
          },
          [divider && <Divider />, $slots.default],
        )}
      </transition>
    );
  },
});

export default Bottom;
