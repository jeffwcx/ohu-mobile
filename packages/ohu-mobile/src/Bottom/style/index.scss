@use '../../_styles/variables' as *;
@use "pkg:@ohu-mobile/styles" as *;
@use '../../_styles/mixins' as *;

$bottomPrefixClass: #{$prefix}bottom;

.#{$bottomPrefixClass} {
  width: 100%;
  position: fixed;
  bottom: 0;
  left: 0;
  @include safe-bottom();
}
