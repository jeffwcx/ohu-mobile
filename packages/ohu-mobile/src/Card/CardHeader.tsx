import { $prefix } from '../_config/variables';
import { defineComponent, props } from '../_utils/defineComponent';

export const cardHeaderBaseName = `${$prefix}card-header`;

const CardHeader = defineComponent('card-header').create({
  props: {
    status: props
      .ofStringLiterals('error', 'success', 'normal')
      .default('normal'),
    bold: props(Boolean).default(false),
    extra: String,
  },
  render() {
    const { $slots, extra, status, bold } = this;
    const extraContent = $slots.extra || extra;
    const root = this.$rootCls()
      .is(status)
      .is(bold && 'bold');
    return (
      <div class={root}>
        {$slots.default}
        {extraContent && (
          <div class={root.element('extra')}>{extraContent}</div>
        )}
      </div>
    );
  },
});

export default CardHeader;
