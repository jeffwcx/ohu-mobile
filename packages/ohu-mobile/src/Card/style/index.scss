@use '../../_styles/variables' as *;
@use "pkg:@ohu-mobile/styles" as *;
@use '../../_styles/mixins' as *;
$cardPrefixClass: #{$prefix}card;
$cardHeaderPrefixClass: #{$cardPrefixClass}-header;

.#{$cardPrefixClass} {
  background: $cardBackground;
  &.has-padding &__content {
    padding: $cardContentPadding;
  }
  &.has-radius {
    border-radius: $cardBorderRadius;
  }
  &.has-shadow {
    box-shadow: $cardBoxShadow;
  }
}

.#{$cardHeaderPrefixClass} {
  position: relative;
  padding: $cardHeaderTopOffset $cardHeaderLeftOffset;
  font-size: $cardHeaderFontSize;
  line-height: 1.375em;
  color: $colorTextBase;
  font-weight: 400;
  &__extra {
    @include center(y);
    right: $cardHeaderLeftOffset;
    color: $cardHeaderExtraColor;
    font-size: $cardHeaderExtraFontSize;
  }
  &.is-bold {
    font-size: $cardHeaderBoldFontSize;
    line-height: 1.38888889em;
    font-weight: bold;
  }
  // ================== state ==================
  &.is-error {
    .#{$cardHeaderPrefixClass}__extra {
      color: $cardHeaderExtraErrorColor;
    }
  }
  &.is-success {
    .#{$cardHeaderPrefixClass}__extra {
      color: $cardHeaderExtraSuccessColor;
    }
  }
}
