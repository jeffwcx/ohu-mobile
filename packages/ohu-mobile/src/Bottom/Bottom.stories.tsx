import Vue from 'vue';
import Bottom from './index';
import './style';
import Button from '../Button';
import '../Button/style';
import Tabbar from '../Tabbar';
import '../Tabbar/style';
import { GovernmentFilled, HomeFilled } from '@ohu-mobile/icons';

import docs from './README.md?raw';

export default {
  title: 'Components/Utils/Bottom',
  parameters: {
    component: Bottom,
    options: {
      showPanel: true,
    },
    notes: { markdown: docs },
  },
};

export const Basic = () =>
  Vue.extend({
    data() {
      return {
        show: true,
      };
    },
    render() {
      return (
        <div style="height: 200vh">
          <Button type="primary" onClick={() => (this.show = !this.show)}>
            toggle
          </Button>
          <Bottom visible={this.show}>
            <Tabbar value={'1'}>
              <Tabbar.Item name="1" icon={HomeFilled}>
                首页
              </Tabbar.Item>
              <Tabbar.Item name="2" icon={GovernmentFilled}>
                政府
              </Tabbar.Item>
            </Tabbar>
          </Bottom>
        </div>
      );
    },
  });

export const Tag = () =>
  Vue.extend({
    render() {
      return (
        <div style="height: 200vh">
          <Bottom tag="footer">
            <Tabbar value={'1'}>
              <Tabbar.Item name="1" icon={HomeFilled}>
                首页
              </Tabbar.Item>
              <Tabbar.Item name="2" icon={GovernmentFilled}>
                政府
              </Tabbar.Item>
            </Tabbar>
          </Bottom>
        </div>
      );
    },
  });
