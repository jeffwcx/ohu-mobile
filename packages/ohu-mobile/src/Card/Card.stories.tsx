import Card from './index';
import './style';
import Button from '../Button';
import '../Button/style';
import DetailItem from '../DetailItem';
import '../DetailItem/style';
import docs from './README.md?raw';

export default {
  title: 'Components/DataDisplay/Card',
  parameters: {
    component: Card,
    options: {
      showPanel: true,
    },
    notes: { markdown: docs },
  },
};

export const Basic = () => ({
  render() {
    return (
      <div style="padding: 10px; background: #F5F5F5;">
        <Card style="margin-bottom: 10px;">simple card</Card>
      </div>
    );
  },
});

export const WithHeader = () => ({
  render() {
    return (
      <div style="padding: 10px; background: #F5F5F5;">
        <Card divider>
          <Card.Header extra="extra">title</Card.Header>
          simple content
        </Card>
      </div>
    );
  },
});

export const HeaderStatus = () => ({
  render() {
    return (
      <div style="padding: 10px; background: #F5F5F5;">
        <Card divider style="margin-bottom: 10px;">
          <Card.Header extra="已取消">订单状态</Card.Header>
          <DetailItem
            title="预约医生"
            content="芬兰 副主任医生"
            unactive
          ></DetailItem>
        </Card>
        <Card divider style="margin-bottom: 10px;">
          <Card.Header status="error" extra="未支付">
            订单状态
          </Card.Header>
          <DetailItem
            title="预约医生"
            content="芬兰 副主任医生"
            unactive
          ></DetailItem>
        </Card>
        <Card divider style="margin-bottom: 10px;">
          <Card.Header status="success" extra="已支付">
            订单状态
          </Card.Header>
          <DetailItem
            title="预约医生"
            content="芬兰 副主任医生"
            unactive
          ></DetailItem>
        </Card>
      </div>
    );
  },
});

export const BoldHeader = () => ({
  render() {
    return (
      <div style="padding: 10px; background: #F5F5F5;">
        <Card divider style="margin-bottom: 10px;" borderless>
          <Card.Header status="success" bold>
            常用工具
            <Button link slot="extra" size="sm">
              更多&gt;{' '}
            </Button>
          </Card.Header>
          这就是内容部分了。
        </Card>
      </div>
    );
  },
});

export const WithShadow = () => ({
  render() {
    return (
      <div style="padding: 10px; background: #F5F5F5;">
        <Card divider style="margin-bottom: 10px;" shadow>
          <Card.Header status="success" extra="已支付">
            订单状态
          </Card.Header>
          <DetailItem
            title="预约医生"
            content="芬兰 副主任医生"
            unactive
          ></DetailItem>
        </Card>
      </div>
    );
  },
});

export const NoPadding = () => ({
  render() {
    return (
      <div style="padding: 10px; background: #F5F5F5;">
        <Card divider style="margin-bottom: 10px;" shadow padding={false}>
          <Card.Header status="success" extra="已支付">
            订单状态
          </Card.Header>
          you see
        </Card>
      </div>
    );
  },
});
