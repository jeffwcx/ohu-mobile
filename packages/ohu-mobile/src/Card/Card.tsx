import { getVNodesByName } from '../_utils/vnode';
import { cardHeaderBaseName } from './CardHeader';
import Divider from '../Divider';
import { CardProps } from './types';
import { defineComponent, props } from '../_utils/defineComponent';

const Card = defineComponent<CardProps>('card').create({
  props: {
    shadow: props(Boolean).default(false),
    borderless: props(Boolean).default(false),
    divider: props(Boolean).default(false),
    padding: props(Boolean).default(true),
  },
  render() {
    const root = this.$rootCls();
    const { $slots, divider } = this;
    root.has([
      !this.borderless && this.shadow && 'shadow',
      !this.borderless && 'radius',
      this.padding && 'padding',
    ]);
    // detect card-header, extra other vnode as content
    if ($slots.default instanceof Array) {
      const headerNodes = getVNodesByName($slots.default, cardHeaderBaseName);
      if (headerNodes.length > 0) {
        const contentNodes = getVNodesByName($slots.default, (name) => {
          return name !== cardHeaderBaseName;
        });
        return (
          <div class={root}>
            {headerNodes}
            {divider && <Divider></Divider>}
            {contentNodes.length > 0 && (
              <div class={root.element('content')}>{contentNodes}</div>
            )}
          </div>
        );
      }
    }
    return (
      <div class={root}>
        <div class={root.element('content')}>{$slots.default}</div>
      </div>
    );
  },
});

export default Card;
