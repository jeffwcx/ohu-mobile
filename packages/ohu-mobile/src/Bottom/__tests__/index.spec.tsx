import { mount, config } from '@vue/test-utils';
import Bottom from '..';
import Card from '../../Card';
import { describe, it, expect } from 'vitest';

config.stubs!.transition = false;

describe('Bottom', () => {
  it('DOM structure', () => {
    const wrapper = mount(Bottom, {
      slots: {
        default: [Card],
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
