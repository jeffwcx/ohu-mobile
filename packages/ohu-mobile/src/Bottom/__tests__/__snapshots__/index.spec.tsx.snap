// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Bottom > DOM structure 1`] = `
<div
  class="ohu-bottom"
>
  <div
    class="ohu-card has-radius has-padding"
  >
    <div
      class="ohu-card__content"
    />
  </div>
</div>
`;

exports[`Bottom DOM structure 1`] = `
<div
  class="ohu-bottom"
>
  <div
    class="ohu-card has-radius has-padding"
  >
    <div
      class="ohu-card__content"
    />
  </div>
</div>
`;
